commands/registrySettings.ts

```diff
@@ -0,0 +1,36 @@
+import * as vscode from 'vscode';
+import { AzureRegistryNode } from "../explorer/models/azureRegistryNodes";
+import { CustomRegistryNode } from "../explorer/models/customRegistryNodes";
+import { DockerHubOrgNode } from "../explorer/models/dockerHubNodes";
+
+export async function setRegistryAsDefault(node: CustomRegistryNode | AzureRegistryNode | DockerHubOrgNode): Promise<void> {
+    let registryName: string;
+    if (node instanceof DockerHubOrgNode) {
+        registryName = node.namespace;
+    } else if (node instanceof AzureRegistryNode) {
+        registryName = node.registry.loginServer || node.label;
+    } else { //CustomREgistryNode
```

Comment:
Do an actual if here, and assert on the else.  Otherwise you won't get notified if a new case gets added to package.json but not to here.

Same for switch's - you should always assert on default.